src/Common/FitOnFhir.Common/Services/QueueService.cs

```diff
@@ -0,0 +1,80 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Azure;
+using Azure.Storage.Queues;
+using EnsureThat;
+using FitOnFhir.Common.Config;
+using FitOnFhir.Common.Interfaces;
+using FitOnFhir.Common.Models;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace FitOnFhir.Common.Services
+{
+    public class QueueService : IQueueService
+    {
+        private readonly ILogger<QueueService> _logger;
+        private readonly QueueClient _queueClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QueueService"/> class.
+        /// Creates a <see cref="QueueClient"/> from the <see cref="AzureConfiguration"/>.StorageAccountConnectionString and <see cref="Constants"/>.QueueName
+        /// </summary>
+        /// <param name="azureConfiguration">The <see cref="AzureConfiguration"/>
+        /// which contains the environment variable value for the storage account connection string.</param>
+        /// <param name="queueClient">An optional <see cref="QueueClient"/> that may be passed in, primarily for testing purposes.</param>
+        /// <param name="logger">An instance of a logger for this class.</param>
+        public QueueService(AzureConfiguration azureConfiguration, QueueClient queueClient, ILogger<QueueService> logger)
+        {
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+
+            if (queueClient == null)
+            {
+                var connectionString = EnsureArg.IsNotEmptyOrWhiteSpace(azureConfiguration.StorageAccountConnectionString);
+                QueueClientOptions queueOptions = new () { MessageEncoding = QueueMessageEncoding.Base64 };
+                _queueClient = new QueueClient(connectionString, Constants.QueueName, queueOptions);
+            }
+            else
+            {
+                _queueClient = EnsureArg.IsNotNull(queueClient, nameof(queueClient));
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task SendQueueMessage(string userId, string platformUserId, string platformName, CancellationToken cancellationToken)
+        {
+            await InitQueue();
```

Comment:
I think you might want this to return a bool (maybe TryInitQueue()), and only run the code below if the init is a success. My concern here is that if CreateIfNotExistsAsync() fails we can safely assume we cannot add messages to the queue, and should not try.